add guest support (#21929)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/search-api-overview.md

See [Use search display layout](/graph/search-concept-display-layout) for examples.

> ## Guest Search

Reviewer: ```suggestion
## Guest search
```

> The search API allows guest users to search SharePoint or OneDrive items that are shared with them. To view guest users,  in the <a href="https://go.microsoft.com/fwlink/p/?linkid=2074830" target="_blank">Microsoft 365 admin center</a>, in the left nav, expand **Users**, and then choose **Guest users**. 

Reviewer: ```suggestion
The Search API enables guest users to easily search for items within SharePoint or OneDrive that have been shared with them. To access the list of guest users, navigate to the <a href="https://go.microsoft.com/fwlink/p/?linkid=2074830" target="_blank">Microsoft 365 admin center</a>, click **Users*in the left navigation menu, and select **Guest users**. 
```

- The search API does not support xrank for **acronym**,**bookmark**,**message**,**chatMessage**, **event**, **person**, **qna**, or **externalItem**.

> - Guest search does not supported for **acronym**,**bookmark**,**message**,**chatMessage**, **event**, **person**, **qna**, or **externalItem**.

Reviewer: ```suggestion
- Guest search does not support for **acronym**,**bookmark**,**message**,**chatMessage**, **event**, **person**, **qna**, or **externalItem**.
```
